packages/@aws-cdk/aws-rds/lib/instance.ts

```diff
@@ -867,23 +867,38 @@ abstract class DatabaseInstanceSource extends DatabaseInstanceNew implements IDa
    *                if you want to override the defaults
    */
   public addRotationSingleUser(options: RotationSingleUserOptions = {}): secretsmanager.SecretRotation {
+    if (options.secret) {
+      //TODO: check given secret is not master secret of this!
+      //TODO: check given secret is not added already!
+      //TODO: how to generate proper ID?
+      const id = 'AdditionalSecretRotation';
+      return this.newSecretRotation(id, options);
+    } else {
+      return this.addRotationSingleUserMasterSecret(options);
+    }
+  }
+
+  private addRotationSingleUserMasterSecret(options: RotationSingleUserOptions): secretsmanager.SecretRotation {
     if (!this.secret) {
-      throw new Error('Cannot add single user rotation for an instance without secret.');
+      throw new Error('Cannot add single user rotation for an instance without master secret.');
     }
 
     const id = 'RotationSingleUser';
-    const existing = this.node.tryFindChild(id);
-    if (existing) {
-      throw new Error('A single user rotation was already added to this instance.');
+    if (this.node.tryFindChild(id)) {
+      throw new Error('A single user rotation for master secret is already added to this instance.');
     }
 
+    return this.newSecretRotation(id, { ...options, secret: this.secret });
+  }
+
+  private newSecretRotation(id: string, options: RotationSingleUserOptions): secretsmanager.SecretRotation {
     return new secretsmanager.SecretRotation(this, id, {
-      secret: this.secret,
       application: this.singleUserRotationApplication,
       vpc: this.vpc,
       vpcSubnets: this.vpcPlacement,
       target: this,
       ...options,
+      secret: options.secret!!,
```

Comment:
Exactly the same set of comments that I gave in `cluster.ts` apply here as well.